addon/adapters/firebase.js

```diff
@@ -418,42 +418,81 @@ export default DS.Adapter.extend(Waitable, {
     });
 
     return new Promise((resolve, reject) => {
-      var savedRelationships = Ember.A();
+
+
+      var relationshipsToSave = [];
+
+
+      // first we remove all relationships data from the serialized record, we backup the
+      // removed data so that we can save it at a later stage.
       snapshot.record.eachRelationship((key, relationship) => {
-        var save;
-        if (relationship.kind === 'hasMany') {
-          if (serializedRecord[key]) {
-            save = this._saveHasManyRelationship(store, typeClass, relationship, serializedRecord[key], recordRef, recordCache);
-            savedRelationships.push(save);
-            // Remove the relationship from the serializedRecord because otherwise we would clobber the entire hasMany
-            delete serializedRecord[key];
-          }
-        } else {
-          if (this.isRelationshipEmbedded(store, typeClass.modelName, relationship) && serializedRecord[key]) {
-            save = this._saveEmbeddedBelongsToRecord(store, typeClass, relationship, serializedRecord[key], recordRef);
-            savedRelationships.push(save);
+        const data = serializedRecord[key];
+        const isEmbedded = this.isRelationshipEmbedded(store, typeClass.modelName, relationship);
+        const hasMany = relationship.kind === 'hasMany';
+        if ( hasMany || isEmbedded ) {
```

Comment:
remove the padding inside the `()`, to be consistent with the code style elsewhere